src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php

```diff
@@ -165,22 +226,85 @@ public function configureOptions(OptionsResolver $resolver)
             return $em;
         };
 
+        // deprecation note
+        $propertyNormalizer = function (Options $options, $propertyName) {
+            if ($propertyName) {
+                trigger_error('The "property" option is deprecated since version 2.7 and will be removed in 3.0. Use "choice_label" instead.', E_USER_DEPRECATED);
+            }
+
+            return $propertyName;
+        };
+
+        // Invoke the query builder closure so that we can cache choice lists
+        // for equal query builders
+        $queryBuilderNormalizer = function (Options $options, $queryBuilder) {
+            if (is_callable($queryBuilder)) {
+                $queryBuilder = call_user_func($queryBuilder, $options['em']->getRepository($options['class']));
+
+                if (!$queryBuilder instanceof QueryBuilder) {
```

Comment:
this is breaking things for other Doctrine projects reusing the DoctrineType (this is why the logic doing this conversion was in the project-specific classes)